bravado/fido_client.py

```diff
@@ -20,6 +20,17 @@
 from bravado.http_future import FutureAdapter
 from bravado.http_future import HttpFuture
 
+if typing.TYPE_CHECKING:
+    class _FidoStub(typing.Protocol):
+        code = None  # type: int
+        body = None  # type: str
+        reason = None  # type: typing.Text
+        headers = None  # type: typing.Mapping[bytes, typing.List[bytes]]
```

Comment:
This doesn't match the type you're using below, which uses unicode strings instead of bytestrings. I'm wondering why mypy didn't catch this.

Reply:
I would assume that mypy is not helping us out on this conditions because `requests` is not typed.
I have to check whether I can use `typeshed` to have some request typing

Reply:
I checked the code and ...
 * `_FidoStub` defines headers to be `Mapping[bytes, List[bytes]]`: this was deduced from [fido|https://github.com/Yelp/fido/blob/10ac95ed5b0e26adfe342147c084f3d0c3908ae0/fido/common.py#L7]
 * `_FidoStub` is used only in `FidoResponseAdapter`.
    `FidoResponseAdapter.heades` need to be conformant to `IncomingResponse.headers` which returns `Mapping[Text, Text]`.
    The code was already decoding the `bytes` key and getting one (the last) value by decoding it. So the end result would be `Mapping[Text, Text]`. This explains the cast of `CaseInsensitiveDict` to `Mapping[Text, Text]`.
    NOTE: the last cast `cast(Mapping[Text, Text], self._headers)` was not needed so I'm removing it